geceval/evaluator.py

```diff
@@ -0,0 +1,321 @@
+                    original_avg_score = evaluator.get_average_score(original_texts)
+                else:
+                    original_avg_score = 0.0
+
+                corrected_scores = defaultdict(dict)
+
+                for prompt_id in prompt_ids:
+                    for model_name in model_names:
+                        corrected_texts = self._collect_corrected_texts(
+                            data[language], prompt_id=prompt_id, model_name=model_name
+                        )
+                        if (
+                            not use_comparative_metrics
+                            and evaluator.supports_single_texts
+                        ):
+                            corrected_avg, scores = evaluator.get_average_score(
+                                corrected_texts
+                            )
+                        else:
+                            corrected_avg, scores = evaluator.get_average_pair_score(
+                                original_texts, corrected_texts
+                            )
+
+                        corrected_scores[prompt_id][model_name] = corrected_avg
+                        log_screen_file(
+                            f"Language: {language}\t Model: {model_name}\t prompt: {prompt_id}\t metric: {self.evaluators[language][module].get_name()}\t score: {corrected_avg}"
+                        )
+                self._aggregate_prompts(
+                    corrected_scores,
+                    prompt_ids,
+                    model_names,
+                    language,
+                    module,
+                    original_avg_score,
+                    use_comparative_metrics,
+                )
+                self._aggregate_models(
+                    corrected_scores,
+                    prompt_ids,
+                    model_names,
+                    language,
+                    module,
+                    original_avg_score,
+                    use_comparative_metrics,
+                )
+
+    def close(self):
+        for language in self.supported_languages:
+            print(f"Closing evaluators for {language}...")
+            for module in self.evaluators[language].keys():
+                self.evaluators[language][module].close()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-e",
+        "--experiment_output_path",
+        help="JSON data.",
+        default="./data/merged_multillm.json.xz",
+    )
+    args = parser.parse_args()
+    experiment_path = args.experiment_output_path
+
+    evaluator = Evaluator()
+    evaluator.evaluate(
+        experiment_path,
+        use_comparative_metrics=True,
+        prompt_ids=[2],
+        # languages=["it"],
+        # model_names=[
+        #    "gemma2B",
+        #    "tower7B1",
+        #    "tower7B2",
+        #    "tower7B3",
+        #    "tower7B4",
+        #    "tower7B5",
+        # ],
```

Comment:
maybe models names and languages could be added to argparse as well

Reply:
Good idea! done